### PR TITLE
Fix _VALID_URLs even further

### PR DESCRIPTION
**Pull request by stepshal**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

### CODE REVIEW

**Comment by yan12125** [comment]
Hi @stepshal did you use any checker to find those bugs? It would be great to integrate it into youtube-dl.

**Comment by stepshal** [comment]
No, i just used this command: `grep -r "_VALID_URL = r'https" youtube-dl > urls`, that search and prints lines with this pattern from directory and then looked at urls, i probably be able to listed only `something.com` without backslash, but i don't know how.

File: youtube_dl/extractor/openload.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 
 class OpenloadIE(InfoExtractor):
-    _VALID_URL = r'https://openload.(?:co|io)/(?:f|embed)/(?P<id>[a-zA-Z0-9-_]+)'
+    _VALID_URL = r'https://openload\.(?:co|io)/(?:f|embed)/(?P<id>[a-zA-Z0-9-_]+)'
```
Comment (stepshal) [inline youtube_dl/extractor/openload.py]: Should `r'https://` be `r'https?://'? Or if the site is only in https, thats how it should be?

**Comment by TRox1972** [comment]
Another improvement could be to replace `www\.` with `(?:www\.)`. Furthermore, wouldn't it be good to have a script that can check for things like this? Could be integrated with PRs.

**Comment by yan12125** [comment]
> Furthermore, wouldn't it be good to have a script that can check for things like this?

It's great to have, though I'm not sure whether checking regular expressions is easy or not.

File: youtube_dl/extractor/openload.py (hunk 1 above)
Comment (yan12125) [inline youtube_dl/extractor/openload.py]: ```https?://``` is better in this case. openload supports both HTTP and HTTPS.

**Comment by stepshal** [comment]
@TRox1972 are you shure it's not `(?:www\.)?` ?

**Comment by stepshal** [comment]
@yan12125  For now i fixed all cases, check please.

**Comment by dstftw** [comment]
```
...F.............
======================================================================
FAIL: test_no_duplicates (__main__.TestAllURLsMatching)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".\test\test_all_urls.py", line 98, in test_no_duplicates
    '%s should not match URL %r . That URL belongs to %s.' % (type(ie).__name__, url, tc['name']))
AssertionError: True is not false : VimeoUserIE should not match URL 'http://vimeo.com/moogaloop.swf?clip_id=2539741' . That URL belongs to Vimeo.

----------------------------------------------------------------------
Ran 17 tests in 5.449s

FAILED (failures=1)
```

**Comment by stepshal** [comment]
@dstftw don't know that's the problem, so vimeo fixes removed for now.

**Comment by stepshal** [comment]
@dstftw look into it again please.

File: youtube_dl/extractor/nhk.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 
 class NhkVodIE(InfoExtractor):
-    _VALID_URL = r'https?://www3\.nhk\.or\.jp/nhkworld/en/vod/(?P<id>.+?)\.html'
+    _VALID_URL = r'https?://(?:www3\.)?nhk\.or\.jp/nhkworld/en/vod/(?P<id>.+?)\.html'
```
Comment (dstftw) [inline youtube_dl/extractor/nhk.py]: `nhk.or.jp` does not even resolves.

**Comment by stepshal** [comment]
@dstftw thiught about it, deleted.

**Comment by dstftw** [comment]
Cherry-picked in 925194022cd661747771e58bad41e5f7ae118999, thanks.

**Pull request closed without merging**